src/app/x/components/x-icon/XIcon.vue

```diff
@@ -0,0 +1,96 @@
+<template>
```

Comment:
**Question**: What’s you general plan with this component? Is this a long-term thing you intend to keep?

It’s generally a bad idea to have a generic icon component that statically imports all its available icons because that will always include them in the resulting code bundles regardless of whether they’re used.